.github/workflows/build.yaml

```diff
@@ -90,3 +99,23 @@ jobs:
       run: |
         cargo install --path .
         agora --version
+
+    - name: Package
+      id: package
+      if: startsWith(github.ref, 'refs/tags/')
+      shell: bash
+      env:
+        TARGET: ${{ matrix.target }}
+        REF: ${{ github.ref }}
+        OS: ${{ matrix.os }}
+      run: ./bin/package
+
+    - name: Publish
+      uses: softprops/action-gh-release@v0.1.5
+      if: startsWith(github.ref, 'refs/tags/')
+      with:
+        draft: false
+        files: ${{ steps.package.outputs.archive }}
+        prerelease: ${{ !startsWith(github.ref, 'refs/tags/v') }}
```

Comment:
Yeah, this is a bit hacky. Also, we don't want random tags to be pre-releases, do we?

Should we maybe come up with a more specific tag format for releases? For example we could say that only tags in the format of `release-$VERSION` are being released. (I don't really have an urgent need for prereleases right now, so I wouldn't bother thinking about that.)

Reply:
The thinking is like this:

We want a way to be able to package and publish tags to GitHub Releases. (They're not real "releases" in the colloquial sense, that's just what GitHub calls "collections of files associated with tags".)

This is useful for testing that the package and publish workflow is correct, and for making binaries available with some patch, for us or others to try out without needing to build from source.

There are two options for marking a GitHub Release as "not a real release". You can mark it as a draft, or as a prerelease. Prerelease seemed more appropriate to me, because non-release releases will never be "undrafted". So they're not really pre-releases, that's just the best of the bad options for marking archives as not real releases.

I updated the script to be more selective. Now there's a `Prerelease Check` step that makes sure that the git ref actually matches `vN.N.N` and doesn't just start with `v`.

Reply:
I see. Thanks for the explanation. I think the word 'draft' would make slightly more sense, IMO. But as far as I can tell, draft releases aren't even publicly visible, so that would not be that useful for providing binaries for testing purposes. So prereleases it is.